tx-helper.go

```diff
@@ -0,0 +1,150 @@
+package ledger
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spacemeshos/ed25519"
+)
+
+// Transaction info struct
+type txInfo struct {
+	PublicKey []byte
+	NetworkID []byte
+	Type      byte
+	Nonce     uint64
+	To        []byte
+	GasLimit  uint64
+	GasPrice  uint64
+	Amount    uint64
+	Data      []byte
+}
+
+// Convert uint64 value to byte array
+func uint64ToBuf(value uint64) []byte {
+	data := make([]byte, 8)
+	data[0] = byte((value >> 56) & 0xff)
+	data[1] = byte((value >> 48) & 0xff)
+	data[2] = byte((value >> 40) & 0xff)
+	data[3] = byte((value >> 32) & 0xff)
+	data[4] = byte((value >> 24) & 0xff)
+	data[5] = byte((value >> 16) & 0xff)
+	data[6] = byte((value >> 8) & 0xff)
+	data[7] = byte((value) & 0xff)
+	return data
+}
+
+// Load transactoin info from JSON file
+func loadTxInfo(fileName string) (*txInfo, error) {
+	txInfo := &txInfo{}
+	jsonFile, err := os.Open("./test" + string(os.PathSeparator) + fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var data map[string]interface{}
+	json.Unmarshal([]byte(byteValue), &data)
+
+	if field, ok := data["networkId"]; ok {
+		txInfo.NetworkID, err = hex.DecodeString(field.(string))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if field, ok := data["type"]; ok {
+		txInfo.Type = byte(field.(float64))
+	}
+
+	if field, ok := data["nonce"]; ok {
+		txInfo.Nonce = uint64(field.(float64))
+	}
+
+	if field, ok := data["to"]; ok {
+		txInfo.To, err = hex.DecodeString(field.(string))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if field, ok := data["gasLimit"]; ok {
+		txInfo.GasLimit = uint64(field.(float64))
+	}
+
+	if field, ok := data["gasPrice"]; ok {
+		txInfo.GasPrice = uint64(field.(float64))
+	}
+
+	if field, ok := data["amount"]; ok {
+		txInfo.Amount = uint64(field.(float64))
+	}
+
+	if array, ok := data["data"]; ok {
+		txInfo.Data = make([]byte, 0)
+		items := array.([]interface{})
+		for _, item := range items {
+			bin, err := hex.DecodeString(item.(string))
+			if err != nil {
+				return nil, err
+			}
+			txInfo.Data = append(txInfo.Data, bin...)
+		}
+	}
+
+	return txInfo, nil
+}
+
+// Convert transaction info to byte array
+func createTx(txInfo *txInfo) []byte {
+	tx := make([]byte, 0)
+	tx = append(tx, txInfo.NetworkID...)
+	tx = append(tx, txInfo.Type)
+	tx = append(tx, uint64ToBuf(txInfo.Nonce)...)
+	tx = append(tx, txInfo.To...)
+	tx = append(tx, uint64ToBuf(txInfo.GasLimit)...)
+	tx = append(tx, uint64ToBuf(txInfo.GasPrice)...)
+	tx = append(tx, uint64ToBuf(txInfo.Amount)...)
+	if txInfo.Data != nil {
+		tx = append(tx, txInfo.Data...)
+	}
+	tx = append(tx, txInfo.PublicKey...)
+	return tx
+}
```

Comment:
You could save many allocations by preallocating `tx` and using, for example:
```go
binary.BigEndian.PutUint64(tx[0:], txInfo.NetworkID)
tx[8] = txInfo.Type
binary.BigEndian.PutUint64(tx[9:], txInfo.Nonce)
copy(tx[17:], txInfo.To)
```
etc.

To simplify - you can still append `Data` and `PublicKey`.